src/tickit/core/components/component.py

```diff
@@ -71,25 +75,38 @@ def __call__(self) -> Component:
 class BaseComponent(Component):
     """A base class for compnents, implementing state interface related methods."""
 
-    state_consumer: StateConsumer[Input]
-    state_producer: StateProducer[Union[Interrupt, Output]]
+    state_consumer: StateConsumer[Union[Input, StopComponent]]
```

Comment:
Would be nice to have a type alias for `Union[Input, StopComponent]`, perhaps `ReceivedMessage`?

Reply:
I do agree. Do you think it is worth being as explicit as:

```
ComponentProducedMessage = Union[Interrupt, Output, ComponentException]
ComponentRecievedMessage = Union[Input, StopComponent]
```
Since schedulers receive outputs and produce inputs.

Reply:
My inclination would be towards `ComponentInput` & `ComponentOutput` which include `InputChanges` and `OutputChanges`, or something of the sort, respectively. Though now I am taking a proper look at this I'm less sure this is the correct model. Perhaps it would be better to have an `Input` class which is generic over `T` and with fields `target: ComponentID` and `content: T`, where `T` can be `ObservedStateChange` or `StopComponent`. Similarly `Output` would be generic over `T`m with `source: ComponentId` and `content: T`, where `T` can be `StateChange`, `Interrupt` or `ComponentException`

Reply:
We're going to put this stuff into a separate issue so we can play around with it there.